src/pool.rs

```diff
@@ -149,11 +230,37 @@ impl ConnectionPool {
         }
     }
 
-    // Connects to a MongoDB server as defined by the initial configuration.
-    fn connect(&self) -> Result<BufStream<TcpStream>> {
+
+    // Connects to a MongoDB server as defined by the initial
+    #[allow(unreachable_code)] // Suppresses warning for `panic` when ssl is enabled.
+    fn connect(&self) -> Result<BufStream<Stream>> {
         let host_name = &self.host.host_name;
         let port = self.host.port;
-        let stream = BufStream::new(try!(TcpStream::connect((&host_name[..], port))));
-        Ok(stream)
+        let inner_stream = try!(TcpStream::connect((&host_name[..], port)));
+
+        if let Some(SslConfig { ca_file: ref _ca_file,
+                                certificate_file: ref _certificate_file,
+                                key_file: ref _key_file }) = self.ssl {
+            #[cfg(feature = "ssl")]
```

Comment:
Putting the feature attribute above a block is very messy, but it seems to be the most ergonomic way to conditionally enable a chunk of code that wouldn't compile if the feature is disabled. I originally intended to use the `if cfg!(...) { ... }` syntax, but it turns out that `cfg!` just expands to either `true` or `false` at compile time, so the code within the block would still be processed by the compiler.